### PR TITLE
Introduces an action to publish Nightly releases and link it in the README with proper instructions

### LINKED ISSUE #128: Extension gets disabled after restarting Firefox
Filing this one to not forget what @flodolo reported as per https://github.com/mozilla/firefox-translations/pull/123#issuecomment-1050184058

### LINKED ISSUE #130: Make Nightly builds easily accessible and linked in the readme

### PR DESCRIPTION
**Pull request by andrenatal**
Fixes #130 
Fixes #128

### CODE REVIEW

File: .github/workflows/publish_nightly.yml
Comment (abhi-agg) [inline .github/workflows/publish_nightly.yml:40]: Is there a reason of having the `tag_name` different than the `release_name`?

File: .github/workflows/publish_nightly.yml
Comment (abhi-agg) [inline .github/workflows/publish_nightly.yml:1]: A general query on this file. Does it create a `Nightly` release artifacts from top of `main` branch and replace the old Nightly release artifacts with the newly created ones?

File: .github/workflows/publish_nightly.yml
Comment (abhi-agg) [inline .github/workflows/publish_nightly.yml:4]: Following the README instructions, people will be able to use Nightly builds from tomorrow onwards because the first nightly build will be generated tomorrow (at 0600 UTC). Right?

File: .github/workflows/publish_nightly.yml
Comment (abhi-agg) [inline .github/workflows/publish_nightly.yml:32]: Same here.

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:40]: Just aesthetics.  I personally don't like tags starting with uppercase and don't like releases names starting with lowercase =D

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:1]: Yes, this action will always delete the previous Nightly release and create and publish a new one, so there will always be the latest available.

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:4]: In theory yes, but I'll try to trigger one automatically during the day today after we merge this

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:25]: That won't happen because if the deletion steps fail for some reason, the create-release step subsequent to this will also fail because the release is already existent and the action can't overwrite it.

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:32]: That won't happen because if the deletion steps fail for some reason, the create-release step subsequent to this will also fail because the release is already existent and the action can't overwrite it. 

I did this so the first one could he properly created

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:25]: I did this so the first one could he properly created

**Comment by abhi-agg** [comment]
Sorry. I just tested and after creating the release for the first time using this workflow, the release creation fails for the subsequent occasions (see [here](https://github.com/abhi-agg/firefox-translations/runs/5409661380?check_suite_focus=true)).

File: .github/workflows/publish_nightly.yml
Comment (abhi-agg) [reply .github/workflows/publish_nightly.yml:41]: Okay.

File: .github/workflows/publish_nightly.yml
Comment (andrenatal) [reply .github/workflows/publish_nightly.yml:23]: To retrieve the tag id so it could be deleted in the next step, as stated in the script

**Comment by andrenatal** [comment]
> Sorry. I just tested and after creating the release for the first time using this workflow, the release creation fails for the subsequent occasions (see [here](https://github.com/abhi-agg/firefox-translations/runs/5409661380?check_suite_focus=true)).

yes, of course it will fail because the action is trying to delete the tag and the release in the *mozilla* branch (just note the call in the action log `https://api.github.com/repos/mozilla/firefox-translations/releases/tags/nightly`), and then in the next step it also fail again because the create-release step is trying to create a release in your *own* branch. 

So if you want to run this in your own branch, you'll need to change the calls to the rest api to point to your own branch: `https://api.github.com/repos/abhi-agg/firefox-translations/releases/tags/nightly` instead of  `https://api.github.com/repos/mozilla/firefox-translations/releases/tags/nightly`

**Review by eu9ene — APPROVED** [review]
lgtm, but we need to test the CI after merge and test that all steps in readme work

**Comment by abhi-agg** [comment]
> > Sorry. I just tested and after creating the release for the first time using this workflow, the release creation fails for the subsequent occasions (see [here](https://github.com/abhi-agg/firefox-translations/runs/5409661380?check_suite_focus=true)).
> 
> yes, of course it will fail because the action is trying to delete the tag and the release in the _mozilla_ branch (just note the call in the action log `https://api.github.com/repos/mozilla/firefox-translations/releases/tags/nightly`), and then in the next step it also fail again because the create-release step is trying to create a release in your _own_ branch and it can't because it already exists since it couldn't be deleted in the previous step.
> 
> So if you want to run this in your own branch, you'll need to change the calls to the rest api to point to your own branch: `https://api.github.com/repos/abhi-agg/firefox-translations/releases/tags/nightly` instead of `https://api.github.com/repos/mozilla/firefox-translations/releases/tags/nightly`

Ahh. Sorry. This was a miss.

**Review by abhi-agg — APPROVED** [review]

**Pull request merged**
